src/main/java/mekanism/client/gui/GuiEmbeddedGaugeTile.java

```diff
@@ -0,0 +1,57 @@
+package mekanism.client.gui;
+
+import mekanism.client.render.MekanismRenderer;
+import mekanism.client.render.MekanismRenderer.FluidType;
+import mekanism.common.tile.prefab.TileEntityContainerBlock;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.inventory.Container;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.fluids.FluidStack;
+
+public abstract class GuiEmbeddedGaugeTile<TILE extends TileEntityContainerBlock> extends GuiMekanismTile<TILE> {
+
+    protected GuiEmbeddedGaugeTile(TILE tile, Container container) {
+        super(tile, container);
+    }
+
+    protected abstract ResourceLocation getGaugeResource();
+
+    protected void displayGauge(int xPos, int yPos, int scale, FluidStack fluid) {
+        displayGauge(xPos, yPos, scale, fluid, 0);
+    }
+
+    protected void displayGauge(int xPos, int yPos, int scale, FluidStack fluid, int side /*0-left, 1-right*/) {
+        if (fluid == null) {
+            return;
+        }
+        int guiWidth = (width - xSize) / 2;
+        int guiHeight = (height - ySize) / 2;
+        int start = 0;
+        int color = fluid.getFluid().getColor(fluid);
+        if (color != -1) {
+            MekanismRenderer.color(color);
```

Comment:
better to avoid `MekanismRenderer.color/resetColor` and use `GlStateManager` instead. (we need to cleanup the direct GL calls, but it wasn't a simple find/replace when I last tried)